### PR TITLE
Refactor Contact Form

### LINKED ISSUE #133: Contact page
# Description
We need to implement the [contact page](https://mst-ka.org/views/contact) to take info sheets from PNMs

# Requirements
- Implement existing input fields
- Implement reCAPTCHA to prevent bots from spamming us (don't enable submit button until reCAPTCHA success)
- Gather & organize data to utilize the existing firebase function to send email to rush chairs.
- Update Readme with any necessary info regarding the implementation of the contact form.

# Reference
image

# Misc Info
I hear dealing with forms in react is hard, luckily there is a popular tool called [Formik](https://formik.org/) that is widely used in the industry that might be a good route to go for us. They even have [integrations with MUI](https://formik.org/docs/examples/with-material-ui)! Do some research on it and determine the best solution for handling this form with react.

### Target Branch: [integration/react-refactor](https://github.com/joe-studer-18/mst-ka-site/tree/integration/react-refactor)

### PR DESCRIPTION
**Pull request by jaredwhanisch**
This is the first stab at refactoring the contact form. I have updated the readme with important information under the **Getting Started** section about initializing Firebase properly, as well as, adding a new process for making PRs, using [hosting channels](https://firebase.google.com/docs/hosting/manage-hosting-resources), so that we aren't overwriting each other on https://react-mst-ka.web.app/ .

To test this I **_highly recommend_** checking out this branch and following the steps listed in the **Utilizing Firebase Emulators to test our Contact Form** section of the readme that I added.

You can see the form and mess around with it here: https://mst-ka--contact-page-refactor-phin3s5l.web.app/contact on a hosting channel. You won't be able to submit anything due to the domain for the hosting channel not being added to one of the ReCAPTCHA domains. 

Let me know if you have any questions!

closes #133

### CODE REVIEW

File: functions/index.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+"use strict";
+
+const functions = require("firebase-functions");
+const admin = require("firebase-admin");
+const nodemailer = require("nodemailer");
+
+admin.initializeApp();
+
+//TODO: Update contactEmails to include the IV, Recruitment chairs, Joe, and Jared, before deploying to prod
+const contactEmails = ["Jared Hanisch <jared.hanisch@gmail.com>"];
```
Comment (joe-studer-18) [inline functions/index.js:16]: nit: can you just update this to include us now? No one will be using it, so it should be safe.

File: utils/unitedStates.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+//Array of United States abbreviations for contact form state dropdown
```
Comment (joe-studer-18) [inline utils/unitedStates.js:1]: Is this really necessary? I know it doesn't take much to maintain (hopefully haha), but does it provide enough value over just asking folks to include a two letter string & passing it through a simple validation regex (if we even want to validate)?

**Review by joe-studer-18 — CHANGES_REQUESTED** [review]

File: functions/index.js (hunk 1 above)
Comment (jaredwhanisch) [reply functions/index.js]: [Fixed](https://github.com/joe-studer-18/mst-ka-site/pull/142/commits/a00257da8612e7ca26bd05d03c83455266be1852#diff-6c53bf76483f89e4e126787e25a40c3d7001cf6590b587c621e2e462cdbc8060R15)

File: utils/unitedStates.js (hunk 2 above)
Comment (jaredwhanisch) [reply utils/unitedStates.js]: [Removed](https://github.com/joe-studer-18/mst-ka-site/pull/142/commits/a00257da8612e7ca26bd05d03c83455266be1852#diff-971e54c897a9db12e59a19bf3c3d10fc32b85f2ad2b20803ed866df0975cb3ad)

**Review by joe-studer-18 — APPROVED** [review]

**Pull request merged**